src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/DisplayNameAuditingEntityListener.kt

```diff
@@ -0,0 +1,30 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.app.database.jpa
+
+import jakarta.persistence.PrePersist
+import jakarta.persistence.PreUpdate
+
+/**
+ * JPA Entity Listener that sets fields annotated with [AuditDisplayName] with the authenticated user's display name.
+ */
+class DisplayNameAuditingEntityListener {
+
+  @PrePersist
+  @PreUpdate
+  fun updateAuditDisplayNameFields(target: Any) {
+    val currentUserDisplayName = UserPrincipalAuditorAware.getCurrentAuditorDisplayName()
+    target.javaClass.declaredFields
+      .filter { field -> field.isAnnotationPresent(AuditDisplayName::class.java) }
```

Comment:
As per my other comment, do we not want separate annotations for create and update?